### PR TITLE
[FIX] uom in lot m2m with serial

### PR DESCRIPTION
**Pull request by ajf-odoo**
[FIX] stock: uom in lot m2m with serial

Consider move uom different from serial product uom
to set up move lines with product uom

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by amoyaux** [comment]
Add the module name in commit message.

File: addons/stock/models/stock_move.py [hunk 1]
```diff
@@ -866,15 +867,14 @@ def _onchange_lot_ids(self):
...
-
         counter = self.env['stock.move.line'].search_count([
             ('company_id', '=', self.company_id.id),
             ('product_id', '=', self.product_id.id),
             ('move_id', '=', self._origin.id),
             ('lot_id', '=', False),
             ('lot_name', '!=', False),
         ])
-        self.update({'quantity_done': len(self.lot_ids) + counter})
+        self.update({'quantity_done': (len(self.lot_ids) + counter)*self.product_uom.factor})
```
Comment (amoyaux) [inline addons/stock/models/stock_move.py:877]: You should rather transform each move line to move UoM. with _compute_quantity and sum the result.

Comment (amoyaux) [reply addons/stock/models/stock_move.py:877]: check _quantity_done_compute in stock_move.py

**Comment by amoyaux** [comment]
Need a rebase to remove ci/template error. Also add a test for the usecase

File: addons/stock/models/stock_move.py [hunk 2]
```diff
@@ -859,22 +860,16 @@ def onchange_product_id(self):
 
     @api.onchange('lot_ids')
     def _onchange_lot_ids(self):
+        qd = sum(ml.product_uom_id._compute_quantity(ml.qty_done,self.product_uom,False) for ml in self.move_line_ids.filtered(lambda x: x.lot_id.id==False and x.lot_name!=False))
```
Comment (amoyaux) [inline addons/stock/models/stock_move.py:863]: space after, and operator like == or !=. Try to install flake8 for python. https://pypi.org/project/flake8/
In addition you can find the error code to ignore there https://www.odoo.com/documentation/13.0/reference/guidelines.html#pep8-options

**Comment by Whenrow** [comment]
robodoo r+

**Pull request closed without merging**